ocamltest/ocaml_actions.ml

```diff
@@ -1050,28 +992,25 @@ let run_test_program_in_toplevel (toplevel : Ocaml_toplevels.toplevel) log env =
             if ocaml_script_as_argument then testfile else "";
             Environments.safe_lookup Builtin_variables.arguments env
           ] in
-          let exit_status =
+          let result =
             if ocaml_script_as_argument
             then Actions_helpers.run_cmd
               ~environment:default_ocaml_env
               ~stdout_variable:compiler_output_variable
               ~stderr_variable:compiler_output_variable
+              ~expected_exit_status
+              ~what
               log env commandline
             else Actions_helpers.run_cmd
               ~environment:default_ocaml_env
               ~stdin_variable:Builtin_variables.test_file
               ~stdout_variable:compiler_output_variable
               ~stderr_variable:compiler_output_variable
+              ~expected_exit_status
+              ~what
               log env commandline
```

Comment:
(Not related to the PR itself, but) it took me some time to notice the difference between the two invocations (the `stdin_variable` variable is not the same).

Reply:
Ah, this is in fact fixed by the next commit in the PR.